clustered/server/build.gradle

```diff
@@ -17,18 +17,18 @@
 apply plugin: EhDeploy
 
 dependencies {
-  compile project(':clustered:common'), "org.slf4j:slf4j-api:$slf4jVersion"
-  compile("org.terracotta:offheap-resource:$terracottaPlatformVersion") {
+  api "org.terracotta:client-message-tracker:$terracottaPlatformVersion"
+  api("org.terracotta:offheap-resource:$terracottaPlatformVersion") {
     transitive = false
   }
-  compile group: 'org.terracotta', name: 'offheap-store', version: offheapVersion
-  compile group: 'org.slf4j', name: 'slf4j-api', version: slf4jVersion
-  compile("org.terracotta.management:monitoring-service-api:$terracottaPlatformVersion") {
+  implementation project(':clustered:common')
+  implementation group: 'org.terracotta', name: 'offheap-store', version: offheapVersion
+  implementation group: 'org.slf4j', name: 'slf4j-api', version: slf4jVersion
+  implementation ("org.terracotta.management:monitoring-service-api:$terracottaPlatformVersion") {
     transitive = false
```

Comment:
What is the transitive for?

Reply:
At a guess I'd say its because this should be declared `compileOnly`... but maybe @anthonydahanne can shed some light?

Reply:
I'm also not sure; it reminds me of maven when you wanted only the present module to "see" the dependency; and not its "embedders".
That was added by @mathieucarbou , who, at this time, probably did not know about [compileOnly](https://blog.gradle.org/introducing-compile-only-dependencies)

Reply:
My little finger is telling me we should go for compileOnly.

Reply:
Looking at the pom I think what is meant here is that this jar is 'implementation', but that it's transitive dependencies are provided:

```xml
<dependencies>
<!--  all these libs will be in common server classpath  -->
<dependency>
<groupId>org.terracotta.management</groupId>
<artifactId>management-model</artifactId>
<version>${project.version}</version>
</dependency>
<dependency>
<groupId>org.terracotta.management</groupId>
<artifactId>management-registry</artifactId>
<version>${project.version}</version>
</dependency>
<dependency>
<groupId>org.terracotta.management</groupId>
<artifactId>sequence-generator</artifactId>
<version>${project.version}</version>
</dependency>
<dependency>
<groupId>org.terracotta.management</groupId>
<artifactId>cluster-topology</artifactId>
<version>${project.version}</version>
</dependency>
<dependency>
<groupId>org.terracotta</groupId>
<artifactId>statistics</artifactId>
<version>${statistics.version}</version>
</dependency>
<!--  provided by voltron (server)  -->
<dependency>
<groupId>org.terracotta</groupId>
<artifactId>entity-server-api</artifactId>
<scope>provided</scope>
</dependency>
<dependency>
<groupId>org.terracotta</groupId>
<artifactId>standard-cluster-services</artifactId>
<scope>provided</scope>
</dependency>
</dependencies>
```
I think maybe this should be implementation and the upstream pom should be fixed instead? @anthonydahanne @mathieucarbou what say you?

Reply:
> upstream pom should be fixed
how ?

Reply:
I'm wondering if all those things should be declared as provided.

Reply:
@anthonydahanne @chrisdennis @henri-tremblay : I think there is a difference between `compileOnly` and `transitive=false`
- `compileOnly` : is when the module needs a dependency but only for compilation time - we do not need for example to have this dependency at runtime (because it is optional or provided) or we do not want this module to transitively bring it in a client project. Examples: logback, or some annotation jars, `entity-server-api`
- `transitive=false`: this is a normal transitive dependency, so needed at runtime and compile time, but only this dependency will come, not the dependencies of this dependency.

Reply:
@mathieucarbou so what bucket does `monitoring-service-api` fall in to?